### ISSUE TITLE
3.5.1 RC2 2D Navigation Regression - TileMap navigation areas no longer being merged

### ISSUE THREAD

**Issue by firebelley** [godotengine/godot#66235]
### Godot version

3.5.1.rc2.mono

### System information

Windows 11

### Issue description

Compared to 3.5.stable.mono, it appears the `Navigation2DServer` no longer combines `TileMap` navigation areas at runtime when `bake_navigation` is true.

In my game, I spawn in additional TileMaps at runtime and programmatically set `bake_navigation` to `true` when they are ready for pathfinding. I have my 32x32 tiles configured to each have a 16x16 NavigationPolygons. I've also configured `navigation/2d/default_edge_connection_margin` in the project settings to be `16` such that these NavigationPolygons will be merged together.

There is this error when adding some TileMaps, which was not appearing before:

```
E 0:00:15.890   sync: Attempted to merge a navigation mesh triangle edge with another already-merged edge. This happens when the current `cell_size` is different from the one used to generate the navigation mesh. This will cause navigation problems.
  <C++ Source>  modules/navigation/nav_map.cpp:582 @ sync()
```
 
Pathfinding worked completely fine when configured this way in 3.5.stable.mono. To be clear, I have made no other changes to the project aside from upgrading Godot to 3.5.1.rc2.mono.

The videos below show the behavior. The orange lines are the paths being drawn. You can see clearly that in 3.5.stable pathing works as intended, while in 3.5.1.rc2 the path is being generated far from the player, at the bottom of the screen.

Interestingly, the orange lines drawn in the 3.5.1.rc2 version are drawn right along the border between where 2 TileMaps meet.

### Current Behavior (3.5.1.rc2)

https://user-images.githubusercontent.com/4622925/191616632-126f9708-3dc6-4f2a-bba6-665036cbc150.mp4

### Expected Behavior (3.5.stable)

https://user-images.githubusercontent.com/4622925/191616653-34b95b91-2dbf-4f00-87e0-fa8a2012b9b6.mp4

I suspect this regression may have been introduced by #64013 which under-the-hood appears to use `get_transform` when previously it would use `get_relative_transform_to_parent`. I am almost positive that those 2 functions return very different results depending on the location of the tilemap in the game. But that is my educated guess after looking through the navigation related changes in this version.

### Steps to reproduce

1. Create a tilemap with navigation
2. Create an agent that can utilize tilemap pathfinding
3. At runtime, start with tilemap `bake_navigation` off.
4. At runtime, turn `bake_navigation` on.
5. At runtime, instance another tile map in a position such that it borders the existing tile map
6. Turn `bake_navigation` on for the instanced TileMap
7. Observe that regions appear not to be connected

### Minimal reproduction project
[Navigation Issue 66235.zip](https://github.com/godotengine/godot/files/9621836/Navigation.Issue.66235.zip)

**Comment by Calinou** [comment]
Can you test this in [3.5.1.rc1](https://godotengine.org/article/release-candidate-godot-3-5-1-rc-1)?

After doing this, if you can [compile the engine from source](https://docs.godotengine.org/en/stable/development/compiling/index.html), you could look into [bisecting the regression](https://docs.godotengine.org/en/stable/community/contributing/bisecting_regressions.html) to greatly speed up troubleshooting.

**Comment by firebelley** [comment]
@smix8 Yeah the error makes me think that the navigation polygons from different tilemaps are actually being generated in the same spot, rather than at the position of the tiles. This seems to be supported by the fact that the paths that are being generated for the enemies is far from the player position. There is no Navigation2D node in my scene, I am using the new Navigation2DServer with TileMaps set to "bake_navigation". So this is definitely a regression since I wasn't using a Navigation2D node in 3.5 and it behaved as expected. Reproduction steps should be sufficient to reproduce. I can try to find time to do that for an MRP.

@Calinou RC1 is working the same as RC2, still different than the behavior in 3.5 stable. I unfortunately do not have my environment setup to compile from source.

**Comment by firebelley** [comment]
@smix8 sorry to spam but I was just able to whip together a reproduction project. In the project, the character follows the mouse. Pressing spacebar will add an extra tilemap and then turn on `bake_navigation` _after_ .5 seconds, which simulates what's happening in my game. You should be able to see that this allows the character to pathfind to that new section in 3.5 but does not in 3.5.1 RC1 or RC2. Please let me know if I can provide any more details!

**Comment by akien-mga** [comment]
Fixed by #66262.
